api/envoy/extensions/filters/http/stateful_session/v3/stateful_session.proto

```diff
@@ -0,0 +1,54 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.stateful_session.v3;
+
+import "envoy/config/core/v3/extension.proto";
+import "envoy/config/core/v3/health_check.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.stateful_session.v3";
+option java_outer_classname = "StatefulSessionProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Stateful Session Filter]
+// This filter sets an override host based on the extensible session state and updates the
+// session state based on the final selected upstream host.
+// [#extension: envoy.filters.http.stateful_session]
+
+// [#comment:TODO(wbpcode): To support global feature enabled or disabled flag.]
+message StatefulSession {
+  // The expected health status of the host. Suppose a session is assigned to the host X.
+  // However, if the health status of host X does not meet expectations, Envoy will select
+  // new host for the session according to the load balancing algorithm. In other words,
+  // the status of the target host must be in this list for session sticky to take effect.
+  //
+  // .. note::
```

Comment:
Should empty list == all? Otherwise its hard to encoded "all" semantics, since if a new state is added to envoy things may have unexpected behavior

Reply:
🤔 It sounds reasonable.

Reply:
I will update this PR to implemented this semantic.

Reply:
And update the docs as well? :)

Reply:
`If no expected health status is configured then any host status will be accepted by default.`

Comments were updated.

Reply:
Awesome. Thanks!